packages/web-app-json-viewer/src/App.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <div class="oc-json-viewer oc-width-1-1 oc-height-1-1">
+    <div
+      ref="editorRef"
+      class="oc-width-1-1 oc-height-1-1"
+      :class="{ 'jse-theme-dark': darkTheme }"
+    ></div>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, onMounted, PropType, ref, unref } from 'vue'
+import { Resource } from '@ownclouders/web-client'
+import { AppConfigObject, useMessages, useThemeStore } from '@ownclouders/web-pkg'
+import { Content, JSONEditor, Mode } from 'vanilla-jsoneditor'
+import { useGettext } from 'vue3-gettext'
+import 'vanilla-jsoneditor/themes/jse-theme-dark.css'
+
+export default defineComponent({
+  props: {
+    applicationConfig: { type: Object as PropType<AppConfigObject>, required: true },
+    currentContent: {
+      type: String,
+      required: true
+    },
+    isReadOnly: { type: Boolean, required: false },
+    resource: { type: Object as PropType<Resource>, required: true }
+  },
+  setup: (props) => {
+    const editorRef = ref<HTMLElement>()
+    const { $gettext } = useGettext()
+    const { showErrorMessage } = useMessages()
+    const themeStore = useThemeStore()
+    const darkTheme = computed(() => {
+      return themeStore.currentTheme.isDark
+    })
+
+    let content: Content = null
+    try {
+      content = {
+        json: JSON.parse(props.currentContent),
+        text: props.currentContent
+      }
+    } catch (e) {
+      content = {
+        json: undefined,
+        text: props.currentContent
+      }
+      showErrorMessage({ title: $gettext('JSON not properly formatted') })
+    }
+
+    onMounted(async () => {
+      const editor = new JSONEditor({
+        target: unref(editorRef),
+        props: {
+          content,
+          mode: Mode.tree,
+          askToFormat: false,
+          readOnly: true,
+          navigationBar: false
+        }
+      })
+
+      // expand all paths up to 2 levels
+      await editor.expand((path) => path.length < 2)
+    })
+
```

Comment:
Looks like the content is losing reactivity here

Reply:
I was thinking about put it into a watcher, but I don't know any case where this app needs to be reactive? Do you have any real life examples

Reply:
Not right now because we removed search from the app layout but I think it's a bad idea to not implement it correctly only because it's not important right now. If we change it again at some point we'll have to fix 27 apps at once

Reply:
If it would just work, I would made the needed changes 😁 but it ain't appear in the config